spring-kafka/src/main/java/org/springframework/kafka/core/KafkaAdmin.java

```diff
@@ -300,7 +301,7 @@ public final boolean initialize() {
 	private void updateClusterId(Admin adminClient) throws InterruptedException, ExecutionException, TimeoutException {
 		try {
 			this.clusterIdLock.lock();
-			if (this.clusterId != null) {
+			if (this.clusterId == null) {
```

Comment:
The original change is this: https://github.com/spring-projects/spring-kafka/commit/c84d0f7cdcf3c9dde35facfcaf90254fca3007c8.
According to its logic this indeed has to be `== null`.
So, i treat this as a simple omission and schedule for back-port.
Thank you for spotting and fixing!